ci/test-clang-docker.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+set -eu
+
+if [[ ! -d /repo ]]; then
+    echo "[-] This script needs to be run within Docker! Use test-clang.sh instead."
+    exit 1
+fi
+
+CC=/repo/toolchain/bin/clang
+CXX=/repo/toolchain/bin/clang++
+hello_c=/repo/llvm-project/hello.c
+hello_cpp=/repo/llvm-project/hello.cpp
+hello_exe=/tmp/hello
+host_arch=$(uname -m)
+
+# Print clang information, which makes sure it can run
+echo "[+] Testing 'clang --version'"
+"$CC" --version
+
+case "$(source /usr/lib/os-release; echo "$ID")" in
+    arch)
+        musl_cc_flags=(
+            -B /usr/lib/musl/lib     # Scrt1.o, crti.o, crtn.o
+            -I /usr/lib/musl/include # stdio.h
+            -L /usr/lib/musl/lib     # -lc
```

Comment:
This I'm surprised by. Does clang not search /usr/lib for headers, crt files, or libraries?

cc @compnerd @MaskRay

Reply:
I don't see why it would look in subdirectories within `/usr/lib`? I'll double check that these are necessary though.

Reply:
Perhaps this is a convention that glibc based distros use to isolate the musl bits from getting mixed in with glibc bits...

If that's the case, I wonder if it makes sense for us to patch clang to look in these directories explicitly when the triple is *-linux-musl`? Thoughts, @MaskRay @compnerd ?

Reply:
> Perhaps this is a convention that glibc based distros use to isolate the musl bits from getting mixed in with glibc bits...

I think that is the case. These particular paths appear to be specific to Arch Linux for now, I have not had a chance to look at other distributions.

These flags are necessary though; without them, we try to link against glibc (`/lib/../lib64/libc.a`):

https://github.com/nathanchance/containers/runs/6629541321?check_suite_focus=true

Reply:
> These flags are necessary though; without them, we try to link against glibc (/lib/../lib64/libc.a):

I guess even though we have a default target triple that's not `*-gnu`, it's not clear based on the path which libc is libc.a.

Reply:
Right, I am guessing that the current assumption is that if a musl triple is being used, we are on a musl distribution, not a glibc distribution with musl installed?